update-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/ApiUpdateDoiRequestHandlerTest.java

```diff
@@ -254,16 +255,16 @@ private DoiRequestSummary readPublicationDirectlyFromDynamo(UUID id) throws NotF
         return doiRequestsService.fetchDoiRequest(id).orElseThrow();
     }
 
-    private DoiRequestSummary expectedDoiRequestSummary(Publication publication) {
+    private DoiRequestSummary expectedDoiRequestSummary(Publication publication,
+                                                        DoiRequestSummary actualDoiRequestSummary) {
         var includedDoiRequest = new DoiRequest.Builder()
             .withDate(mockNow)
             .withStatus(DoiRequestStatus.APPROVED)
             .build();
-        return new DoiRequestSummary(
-            publication.getIdentifier(),
-            publication.getOwner(),
-            includedDoiRequest,
-            publication.getEntityDescription());
+        publication.setDoiRequest(includedDoiRequest);
+        // copy actual modified date because it is set in Publication class by uncontrollable clock.
+        publication.setModifiedDate(actualDoiRequestSummary.getPublicationModifiedDate());
```

Comment:
You can also assert that the data is within 10 seconds around the test. 
But it makes it hard to do a full assert equals between objects, so we want to be able to inject a fake clock down the line.

Reply:
I think we have to revisit how to add functionality closer to the data model

Reply:
@norrs is there any specific action you would like here?